blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/datastax/DBasicMetricsRW.java

```diff
@@ -0,0 +1,350 @@
+        // if there are strings & booleans in this request, they will be tracked as numeric metrics.
+        // not sure how we want to get around that.
+        Timer.Context ctx = Instrumentation.getWriteTimerContext( CassandraModel.CF_METRICS_FULL_NAME );
+
+        Map<Locator, ResultSetFuture> futures = new HashMap<Locator, ResultSetFuture>();
+
+        try {
+
+            for( IMetric metric : metrics ) {
+
+                if (!shouldPersist(metric)) {
+                    LOG.trace("Metric shouldn't be persisted, skipping insert", metric.getLocator().toString());
+                    continue;
+                }
+
+                Locator locator = metric.getLocator();
+
+                if( !isLocatorCurrent( locator ) ) {
+
+                    setLocatorCurrent( locator );
+
+                    if( !DataType.isStringOrBoolean( metric.getMetricValue() ) )
+                        locatorIO.insertLocator( locator );
+                }
+
+
+                futures.put( locator, rawIO.insertAsync( metric ) );
+            }
+
+            for( Map.Entry<Locator, ResultSetFuture> f : futures.entrySet() ) {
+
+                try {
+                    ResultSet result = f.getValue().getUninterruptibly();
+                    LOG.trace( "result.size=" + result.all().size() );
+                }
+                catch ( Exception e ) {
+                    Instrumentation.markWriteError();
+                    LOG.error(String.format("error writing metric for locator %s",
+                            f.getKey()), e );
+                }
+            }
+        }
+        finally {
+
+            ctx.close();
+        }
+    }
+
+    /**
+     * Fetches {@link com.rackspacecloud.blueflood.outputs.formats.MetricData} objects for the
+     * specified {@link com.rackspacecloud.blueflood.types.Locator} and
+     * {@link com.rackspacecloud.blueflood.types.Range} from the specified column family
+     *
+     * @param locator
+     * @param range
+     * @param gran
+     * @return
+     */
+    @Override
+    public MetricData getDatapointsForRange( final Locator locator, Range range, Granularity gran ) throws IOException {
+
+        Map<Locator, MetricData> result = getDatapointsForRange( new ArrayList<Locator>() {{ add( locator ); }}, range, gran );
+        return result.get( locator );
+    }
+
+    /**
+     * Fetches {@link com.rackspacecloud.blueflood.outputs.formats.MetricData} objects for the
+     * specified list of {@link com.rackspacecloud.blueflood.types.Locator} and
+     * {@link com.rackspacecloud.blueflood.types.Range} from the specified column family
+     *
+     * @param locators
+     * @param range
+     * @param gran
+     * @return
+     */
+    @Override
+    public Map<Locator, MetricData> getDatapointsForRange( List<Locator> locators, Range range, Granularity gran ) throws IOException {
+
+        Timer.Context ctx = Instrumentation.getReadTimerContext( "DBasicMetricsRW.getDatapointsForRange" );
```

Comment:
Oh yeah, its reading that as well.  But its the same question, right?  Despite the granularity you might have strings and you might have numerics.